File: frontTimes/FrontTimes.java
package frontTimes;

public class FrontTimes {
    //Retorne as 3 primeiras letras da String n vezes, ou que ela tiver (caso seja menor que 3), para n >= 0.

    public String frontTimes(String str, int n) {
        StringBuilder stringBuilder = new StringBuilder();
        if (str.length() < 3){
            for (int i = 0; i < n; i++) {
                stringBuilder.append(str);
            }
        } else {
            for (int i = 0; i < n; i++) {
                stringBuilder.append(str.substring(0,3));
            }
        }
        return stringBuilder.toString();
    }

    //ENGENHOSO
    public String frontTimes_2(String str, int n) {
        int frontLen = 3;
        if (frontLen > str.length()) {
            frontLen = str.length();
        }
        String front = str.substring(0, frontLen);

        String result = "";
        for (int i=0; i<n; i++) {
            result = result + front;
        }
        return result;
    }
}

File: Python/Warmup-2/front_times.py
"""
Dada uma String "str" e sabendo que o "front" é considerado as 3 primeiras letras que
a compõem, ou o que quer que ela tenha até lá. Retorne "n" cópias do "front" da string
dada.
"""


def front_times(str, n):
    front = 3
    if len(str) < 3:
        front = len(str)
    return str[:front]*n


print(front_times("SA", 3))
